assets/src/hooks/useDetour.ts

```diff
@@ -73,45 +104,40 @@ export const useDetour = ({ routePatternId, shape }: OriginalRoute) => {
     })
   }
 
-  const canAddWaypoint = () => startPoint !== null && endPoint === null
+  const canAddWaypoint = () =>
+    snapshot.can({
+      type: "detour.edit.place-waypoint",
+      location: { lat: 0, lon: 0 },
```

Comment:
note: if the event has guards which check this value, then we'd poentially need to make this an input to the "can" check function.

Reply:
I'm not sure I understand this comment?

Reply:
I mainly just wanted to mention that the values in `location` are _potentially_ significant. `.can` checks any associated `guards` on an event before reporting if you "can" do something or not. Because a guard can read the event parameters, a guard _could_ deny `location` values that don't meet arbitrary conditions